### PR TITLE
Fix a recent template-specialization-related regression.

### PR DESCRIPTION
**Pull request by emilio**
r? @upsuper

### CODE REVIEW

**Comment by emilio** [comment]
> cpp/TypeTraits.h:3:10: fatal error: 'type_traits' file not found

I'll have to drop the second commit before landing I guess :(

**Comment by upsuper** [comment]
You may need to add `-std=c++11` since `type_traits` is added in C++11.

**Comment by emilio** [comment]
I already do that though, right?

**Comment by upsuper** [comment]
Probably also `-stdlib=libc++`?

**Comment by emilio** [comment]
Let's try, thanks for the suggestion :)

File: bindgen-integration/build.rs [hunk 1]
```diff
@@ -22,6 +22,18 @@ fn main() {
         .generate()
         .expect("Unable to generate bindings");
 
+    // Just tests that parsing <type_traits> doesn't crash, since we've have
```
Comment (fitzgen) [inline bindgen-integration/build.rs]: Nitpick: "we've have" -> "we've **had**"

File: bindgen-integration/build.rs [hunk 2]
```diff
@@ -22,6 +22,18 @@ fn main() {
...
+    // multiple hard-to-reduce regressions in that header.
```
Comment (fitzgen) [inline bindgen-integration/build.rs]: Aside: have you tried creducing the post-preprocessed test case? creduce can usually do a better job than trying to reduce manually.

**Comment by fitzgen** [comment]
Dang :(

> cpp/TypeTraits.h:3:10: fatal error: 'type_traits' file not found

Seems to only be on linux, though. Perhaps we can `if cfg!(target_os = "linux") { ... }` and manually add `/usr/include/c++/<version>/type_traits`? Perhaps leveraging the `glob` crate?

Another possibility is to just pull in the post-preprocessed `type_traits` file into our test suite directly.

**Comment by emilio** [comment]
I wouldn't want to block on that for now, I'll open a new PR to figure out how to make travis happy.

@bors-servo r=fitzgen

File: bindgen-integration/build.rs (hunk 2 above)
Comment (emilio) [inline bindgen-integration/build.rs]: nope, should try

**Pull request merged**